client/src/compute/rest/index.ts

```diff
@@ -0,0 +1,135 @@
+import { RunResult, Session } from "..";
+import { ContextsApi, LogLine } from "../api/compute";
+import { clearTokens, getAccessToken } from "./auth";
+import { ComputeState, getApiConfig } from "../common";
+
+export interface Config {
+  endpoint: string;
+  clientId?: string;
+  clientSecret?: string;
+  context?: string;
+  serverId?: string;
+}
+
+import { ComputeServer } from "../server";
+import { ComputeSession } from "../session";
+import { ComputeJob } from "../job";
+
+let config: Config;
+let computeSession: ComputeSession | undefined;
+
+async function setup() {
+  const apiConfig = getApiConfig();
+  if (!config.serverId) {
+    apiConfig.accessToken = await getAccessToken(config);
+  }
+
+  if (computeSession && computeSession.sessionId) {
+    const state = await computeSession
+      .getState()
+      .catch(() => (computeSession = undefined));
+    if (state === ComputeState.Error) {
```

Comment:
Do you have a test case that enters syntax check mode? I fell like it would be better to be able to detect that mode and only cancel when needed instead of every time.

In the mean time I will change it back to cancel every time and open an issue to look into only sending cancel when necessary.

Reply:
run
```
proc asd;
run;
```
and it will enter syntax check mode